src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -320,11 +369,113 @@ namespace AppInstaller::Repository
             std::shared_ptr<IPackageVersion> m_trackingPackageVersion;
         };
 
+        // Wrapper around an available package to add pinning functionality for composite packages.
+        // Most of the methods are only here for completeness of the interface and are not actually used.
+        struct CompositeAvailablePackage : public IPackage
+        {
+            CompositeAvailablePackage() {}
+            CompositeAvailablePackage(std::shared_ptr<IPackage> availablePackage, std::optional<Pinning::Pin> pin = {})
+                : AvailablePackage(availablePackage), Pin(pin)
+            {
+                auto latestAvailable = AvailablePackage->GetLatestAvailableVersion(PinBehavior::IgnorePins);
+                if (latestAvailable)
+                {
+                    SourceId = latestAvailable->GetSource().GetIdentifier();
+                }
+            }
+
+            std::string SourceId;

```

Comment:
Should the 3 member variables be private? and named m_*?

Reply:
Do you feel strongly about this?
It was easier for me to keep them public as I use those fields in the composite package, and the Pin one I also set from there.

Reply:
If it's not a lot of code changes, I think it'd be better to be private. Because this struct is more of a class and less of a data structure now.

But if it's too many changes, I'm good with leaving it as is. Your call.